lib/src/builder/codegen/accessors_generator.dart

```diff
@@ -259,14 +259,19 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
           docComment = '';
         }
 
+        String castGetterMapValueIfNecessary(String expression) {
+          if (typeSource == null) return expression;
+          return '($expression) as $typeSource';
+        }
+
         String generatedAccessor = '$docComment'
             // TODO reinstate inlining once https://github.com/dart-lang/sdk/issues/36217 is fixed and all workarounds are removed
             // inlining is necessary to get mixins to use $index/$indexSet instead of $index$asx
             // '  @tryInline\n'
             '  @override\n'
             '$metadataSrc'
-            '  ${typeString}get $accessorName => $proxiedMapName[$keyConstantName] ?? null;'
-            ' // Add ` ?? null` to workaround DDC bug: <https://github.com/dart-lang/sdk/issues/36052>;\n'
+            // Add ` ?? null` to workaround DDC bug: <https://github.com/dart-lang/sdk/issues/36052
+            '  ${typeString}get $accessorName => ${castGetterMapValueIfNecessary('$proxiedMapName[$keyConstantName] ?? null')};\n'
```

Comment:
why is `?? null` also being type casted?

Reply:
I figured applying the DDC workaround ASAP would be more robust, so that any potential `undefined` value can't make its way into DDC runtime cast logic and cause issues (for example, if the DDC implementation changes in a newer SDK version and suddenly doesn't work well with `undefined`).

That's a good question, though; I can add that info in a comment!

Reply:
Added in https://github.com/Workiva/over_react/pull/669/commits/79ef12a8c13737cf47cb1999021fc32c0b7e5c7b